### ISSUE TITLE
Investigate using `RegisterEventHotKey` instead of Event taps.

### ISSUE THREAD

**Issue by numist** [numist/Switch#31]
This is a pretty low priority—it would only be useful for listing Switch in the App Store.

**Comment by borismus** [comment]
@numist why closed? Did you find this to be impossible?

**Comment by numist** [comment]
Not at all, but there were other factors at play specific to Switch's purpose.

Two arguments in favour of Switch sticking with an event tap:
- Switch doesn't want to be the active application; it makes application and window order more difficult to maintain, and the Accessibility IPC calls are inherently slow and, sadly, error-prone. At the same time, Switch _does_ want to receive mouse events when it is in use. The event manager implemented in Switch forwards mouse events to the UI when it is being shown, solving both of these problems.
- Ultimately I'd like to support custom keybindings. While this is totally possible with `RegisterEventHotKey`, it is not possible to override `⌘⇥` using anything other than an event tap.

Those two arguments are very application-specific, and there exist at least as many good general arguments _against_ using an event tap:
- Event taps slow down the entire system. It's critically important that the event handler be the fastest part of your application because it will be run for every single input event. Even if your handler is a return statement there is a significant cost—the time spent context switching  adds up. With `RegisterEventHotKey`, you only get events when they are relevant to you.
- Event taps do not work in sandboxed applications. I believe the sandbox to be generally a Good Thing™ for users, and a decent guardrail for developers. Unfortunately, it's already impossible to sandbox Switch due to the use of Accessibility and `CGWindowListCreateImage`.
